pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -196,6 +196,15 @@ func (gc *GarbageCollector) Sync(ctx context.Context, discoveryClient discovery.
 			return
 		}
 
+		// In partial discovery cases, don't remove any existing synced informers for resources in the failed groups, just add new ones
```

Comment:
'just add new ones' seems a little confusing here - we don't _add_ any new ones, we just retain all informers that already exist (i.e. are already in `oldResources`), right?

Reply:
rephrased the comment to be less confusing:

> In partial discovery cases, preserve existing synced informers for resources in the failed groups, so resyncMonitors will only add informers for newly seen resources